test/napi/src/js/threads.rs

```diff
@@ -144,3 +144,44 @@ pub fn leak_event_queue(mut cx: FunctionContext) -> JsResult<JsUndefined> {
 
     Ok(cx.undefined())
 }
+
+pub fn drop_global_queue(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+    struct Wrapper {
```

Comment:
Neat use of nested struct types. If you asked me in an interview question if this was legal I wouldn't have known! 😂 But it's very readable.

Reply:
I agree! I prefer this locality for structs and functions that are for a very specific purpose that aren't intended to be generally usable externally.

If I understand correctly, closures are essentially sugar for this.